File: app/src/main/java/com/anbillon/lunarlite/di/module/ActivityModule.java
package com.anbillon.lunarlite.di.module;

import android.app.Activity;
import android.content.Context;

import com.anbillon.lunarlite.di.scope.PerActivity;

import dagger.Module;
import dagger.Provides;

/**
 * Dagger module that provides activity level related collaborators.
 *
 * @author Vincent Cheung (coolingfall@gmail.com)
 */
@Module public final class ActivityModule {
	private final Activity mActivity;

	public ActivityModule(Activity activity) {
		mActivity = activity;
	}

	@Provides @PerActivity Context provideActivityContext() {
		return mActivity;
	}
}

File: app/src/main/java/com/anbillon/lunarlite/di/module/android/ActivityModule.kt
package com.anbillon.lunarlite.di.module.android

import android.app.Activity
import com.anbillon.lunarlite.di.component.AndroidComponent.LunarComponent
import com.anbillon.lunarlite.ui.lunar.LunarLiteActivity
import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap

/**
 * @author Vincent Cheung (coolingfall@gmail.com)
 */
@Module(subcomponents = arrayOf(LunarComponent::class)) abstract class ActivityModule {
  @Binds
  @IntoMap
  @ActivityKey(LunarLiteActivity::class) internal abstract fun bindAlmanac(
      builder: LunarComponent.Builder): AndroidInjector.Factory<out Activity>

}